pkg/minikube/cluster/cluster.go

```diff
@@ -122,12 +132,27 @@ func DeleteHost(api libmachine.API) error {
 	if err != nil {
 		return err
 	}
+
+	if err = unregister(host); err != nil {
+		return err
+	}
+
 	m := util.MultiError{}
 	m.Collect(host.Driver.Remove())
 	m.Collect(api.Remove(constants.MachineName))
 	return m.ToError()
 }
 
+// Un-register host VM
+func unregister(host *host.Host) error {
+	if drivers.MachineInState(host.Driver, state.Running)() {
```

Comment:
ahh, but without a running VM we cannot test whether the VM is registered, so we cannot know whether and what to report to the user. 

Really, we should not rely on the VM to determine whether it is registered. Instead we probably should write a config entry during start that the VM was registered and check for this entry during deletion. This was we don't have to rely on a running VM to even detect whether un-registration is necessary. WDYT?